### DISCUSSION TITLE
Report sections

### DISCUSSION THREAD

**Discussion by svpguru** [Syslifters/sysreptor#4] [Q&A] [answered]
Hi, what would be the best way to display different sections in the report template for testing types, such as Wireless, Internal Infrastructure and Web Applications sections? thanks

**Comment by aronmolnar** [comment, marked as answer]
Hi,
Currently, findings are sorted by the CVSS score during reporting (this will be more flexible in the future). However, in the PDF report, you can order your findings as you like.

You could create an "enum" finding field that holds your testing types.

In the report, you could iterate through your findings and only select those of a certain type. We do the same in the OSCP exam report, where we select Active Directory findings:

```
<section v-if="findings.find(finding => finding.isActiveDirectory === true)">
    <h1 id="findingsad" class="in-toc numbered">Active Directory Set</h1>

    <div v-for="(finding, index) in findings">
        <div v-if="finding.isActiveDirectory">
            <h2 :id="finding.id" class="in-toc numbered">{{ finding.title }} ({{ finding.ip_address }})</h2>
...
```

**Reply by svpguru** [reply]
Great, thanks

**Comment by svpguru** [comment]
Hey, thanks for the reply above I've just go around to looking at this however I'm lost, I have setup an enum data type with an ID of 'testing_type' I have the set a number of values such as 'internal_inf' 'external_inf' and 'web_app'. Any help on getting these to populate the toc with the findings listed underneath would be greatly appreciated. :)

**Reply by MWedl** [reply]
You can try following snippet. It groups findings with the same `testing_type` and renders separate chapters with a list of findings for each type. The CSS class `in-toc` adds the headlines to the table of contents (this only works on `<h*>` tags).

```html
<section v-for="group in lodash.sortBy(Object.entries(lodash.groupBy(findings, 'testing_type.value')).map(([k, v]) => ({testing_type: k, label: v[0].testing_type.label, findings: v})), ['label'])">
    <h1 :id="'findings-' + group.testing_type" class="in-toc numbered">{{ group.label }}</h1>
    <div v-for="finding in group.findings">
        <h2 :id="finding.id" class="in-toc numbered">{{ finding.title }}</h2>
        <div>
            <strong>Testing Type:</strong> {{ finding.testing_type.label }}
        </div>
    </div>
</section>
```
